### PR TITLE
EquipmentMenu: Fix GetClass error

### PR DESCRIPTION
**Pull request by TimGoll**
This error was reported by @mexikoedi on Discord:

image

I fixed this by correctly using `WEPS.GetClass` here instead of `equipment:GetClass`. While doing this I noticed that items have no equivalent of `GetClass` and `IsValid` so I added that to the base class. These changes are not strictly needed for this fix, but imho they are useful for consistency.

### CODE REVIEW

**Review by Histalek — APPROVED** [review]
LGTM

File: lua/terrortown/menus/gamemode/equipment/base_equipment.lua [hunk 1]
```diff
@@ -179,9 +179,11 @@ function CLGAMEMODESUBMENU:Populate(parent)
         database = DatabaseElement(accessName, itemName, "damageScaling"),
     })
 
+    local equipmentClass = WEPS.GetClass(equipment)
```
Comment (saibotk) [inline lua/terrortown/menus/gamemode/equipment/base_equipment.lua:182]: Isnt this only intended for weapons?

I think this should not be used for items too

File: lua/terrortown/entities/items/item_base/shared.lua [hunk 2]
```diff
@@ -100,9 +100,32 @@ function ITEM:Bought(ply) end
...
 function ITEM:IsEquipment()
     return WEPS.IsEquipment(self)
 end
+
+---
+-- Returns the classname of an item. This is often the name of the Lua file or folder containing the files for the item.
+-- @return string The item's class name
+-- @realm shared
+function ITEM:GetClass()
+    return self.ClassName
```
Comment (saibotk) [inline lua/terrortown/entities/items/item_base/shared.lua:116]: This is very confusing, what is the difference between the class name here and the class name returned by the weps function?

**Review by saibotk — CHANGES_REQUESTED** [review]

File: lua/terrortown/entities/items/item_base/shared.lua (hunk 2 above)
Comment (TimGoll) [reply lua/terrortown/entities/items/item_base/shared.lua:116]: It's identical. But entities (like weapons) have a function called `wep:GetClass()`. For some reason items didn't have that function and only a variable named `.ClassName` (which I think also exists on entites). In our code we mostly handle items and weapons as similar things, so imho it should be good to have the same basic functions.

The weps function basically checks if the equipment is a table and then returns `.ClassName` or returns `:GetClass()` it the equipment `:IsValid()`

File: lua/terrortown/menus/gamemode/equipment/base_equipment.lua (hunk 1 above)
Comment (TimGoll) [reply lua/terrortown/menus/gamemode/equipment/base_equipment.lua:182]: we always use this for both in our code base and it works as well. But yes, the name is misleading

**Review by saibotk — APPROVED** [review]

**Pull request merged**